### ISSUE TITLE
Add a setting to manually set the Pane highlight color

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/terminal#3061]
Follow up from #994 and PR:#3060

In the aforementioned PR, we're just using the accent color for highlighting the pane. We should let that be a customizable global setting.

Questions:
* There's been discussion before about allowing custom theming, hypothetically powered by XAML. Would we want to wait for that discussion to be had before we do this?
  - We'd probably want to change it from using `SystemAccentColor` to some `FocusedPaneBorder` color resource, that we define by default to be `SystemAccentColor`, should we decide to pursue this option
  - If we do want to do this in XAML later, would enabling it in the json _now_ negatively impact that future scenario?

**Comment by zadjii-msft** [comment]
@javierdlg I think you're probably looking for #3326 - this issue is about the color used to indicate that a pane is focused (which is currently in PR in #3060). I'll respond to the ask in that thread :)

**Comment by SkyTrix** [comment]
Not sure if this should be a separate issue, but in addition to this, it would be nice to be able to completely hide the focused pane border when the Terminal isn't the active application. I often have a few panes open on a separate screen where the focused pane border becomes more of a distraction than a help when the Terminal isn't the active window and I'm just glancing over at the outputs of the different panes.

**Comment by zadjii-msft** [comment]
> **Note**
> ## Walkthrough

* [ ] Start by adding a new Theme object for Panes:
  * [ ] `PaneTheme` in `Theme.idl`, quite similar to `TabRowTheme` in the same file
  * [ ] Add a single `ThemeColor ActiveBorderColor` to that `runtimeclass`. 
  * [ ] In `MTSMSettings.h`, define a new `MTSM_THEME_PANE_SETTINGS`, again taking cues from the tab row version. The key for this one should just be `"pane"`
  * [ ] Add a `THEME_OBJECT(PaneTheme, MTSM_THEME_PANE_SETTINGS);` to `Theme.h`
  * [ ] Add the pane theme to the other places theme objects appear in Theme.cpp as well.
* [ ] The "hard" part: actually plumb that color through to the panes
  * [ ] `Pane` currently uses a `static` `Brush for the border colors. It probably shouldn't.
  * [ ] `TerminalPage::_updateThemeColors` is what updates theme colors to the tabs, the window, etc. Similarly, it should plumb that through to all the panes.
    * [ ] `WalkTree` is a really handy helper for doing something to every pane.
  * [ ] Make sure that a `null` or otherwise omitted `pane.activeBorderColor` still uses `accent` by default
* [ ] While you're here, probably add a similar `pane.inactiveBorderColor`, for controlling the appearance of the inactive pane border color. 
    
    
Once this is done, if you wanted to get really ambitious, you could follow up with a similar property to configure the pane hover color, ala #8564.

**Comment by peter-lee20** [comment]
Hello, can I take a stab at this?

**Comment by Haaziq386** [comment]
@zadjii-msft where is the   `Theme.idl` file i.e in which folder ?

**Comment by zadjii-msft** [comment]
@Haaziq386  [src/cascadia/TerminalSettingsModel/Theme.idl](https://github.com/microsoft/terminal/blob/8d3f12b1c065a965974cd6b5bc7a81176ffb0f1b/src/cascadia/TerminalSettingsModel/Theme.idl)

**Comment by light-N-H** [comment]
Hi, I was reading through this PR and the discussion around it. It looks like there was positive feedback on the proposed solution by techypanda , so I was curious whether there are any remaining concerns or blockers preventing it from being merged. 
as i can see this problem is open for anyone to still try and take a stab at? 
Thanks for any context you can share.
